atc/exec/build/repository.go

```diff
@@ -1,71 +1,68 @@
-package artifact
+package build
 
 import (
 	"sync"
 
-	"github.com/concourse/concourse/atc/worker"
+	"github.com/concourse/concourse/atc/runtime"
 )
 
-// Name is just a string, with its own type to make interfaces using it
+// ArtifactName is just a string, with its own type to make interfaces using it
 // more self-documenting.
-type Name string
+type ArtifactName string
 
-// Repository is the mapping from a Name to an ArtifactSource.
+// Repository is the mapping from a ArtifactName to an Artifact.
 // Steps will both populate this map with new artifacts (e.g.  the resource
 // fetched by a Get step), and look up required artifacts (e.g.  the inputs
 // configured for a Task step).
 //
 // There is only one ArtifactRepository for the duration of a build plan's
```

Comment:
Is it still true that ArtifactRepository is an in-memory data structure that lasts only for one build lifetime? I wonder if there's a way to make this relationship between an artifact and a build crystal clear.